include/sdf/Gui.hh

```diff
@@ -66,6 +69,36 @@ namespace sdf
     /// \return SDF element pointer with updated gui values.
     public: sdf::ElementPtr ToElement() const;
 
+    /// \brief Get the number of plugins.
+    /// \return Number of plugins contained in this Gui object.
+    public: uint64_t PluginCount() const;
+
+    /// \brief Get a plugin based on an index.
+    /// \param[in] _index Index of the plugin. The index should be in the
+    /// range [0..PluginCount()).
+    /// \return Pointer to the plugin. Nullptr if the index does not exist.
+    /// \sa uint64_t PluginCount() const
+    public: const Plugin *PluginByIndex(const uint64_t _index) const;
+
+    /// \brief Get whether a plugin name exists.
+    /// \param[in] _name Name of the plugin to check.
+    /// \return True if there exists a plugin with the given name.
+    public: bool PluginNameExists(const std::string &_name) const;
+
+    /// \brief Get a plugin based on a name.
+    /// \param[in] _name Name of the plugin.
+    /// \return Pointer to the plugin. Nullptr if the name does not exist.
+    public: const Plugin *PluginByName(const std::string &_name) const;
```

Comment:
Since plugin names are not necessarily unique (https://github.com/ignitionrobotics/sdformat/pull/721 and https://github.com/ignitionrobotics/sdformat/pull/722), maybe we should only provide `PluginByIndex`?

Reply:
Done in 980d5d2 and updated the SDF spec to remove the `unique` wording.